Modules/rat_rescue.py

```diff
@@ -784,22 +787,33 @@ def rats(self, value):
                      await RatCache().get_rat_by_name(name))
             if found[0]:
                 self.rats.append(found[0])
-                return
+                assigned_rat = found[0]
             elif found[1]:
                 # a generic match (not platform specific) was found
                 # TODO throw a warning so the invoking method can handle this condition
                 log.warning("A match was found, but it was not the right platform!")
                 self.rats.append(found[1])
-                return
+                assigned_rat = found[1]
 
             else:
                 # lets make a new Rat!
-                if self.rat_board:  # PRAGMA: NOCOVER
-                    raise NotImplementedError  # TODO fetch rat from API
+                # if self.rat_board:  # PRAGMA: NOCOVER
+                #    pass  # TODO fetch rat from API
                 # TODO: fetch rats from API handler, use that data to make a new Rat instance
 
                 rat = Rat(name=name, uuid=guid)
                 self.rats.append(rat)
+                assigned_rat = rat
+
+        if isinstance(guid, UUID):
+            # lets check if we already have this rat in the cache
+            found = await RatCache().get_rat_by_uuid(guid)
```

Comment:
"the absence of evidence is not evidence of absence"
 - rat might not be in the cache 😉

Reply:
If the rat is not in the cache, we may need to query the API to see if the Rat exists. at the very least this branch should be accounted for, but unused as that depends on a feature that is not currently merged

Reply:
It's just left unhandled right now...I could just add a TODO comment if that's what you were looking for (currently the comment says not found rats return None, which is the current behavior).

Reply:
i think that would be advisable, since most of us use an IDE that recognizes TODO items. As much as i know todos are ugly, i find thats better than ignoring an obvious missed branch.
That or make another JIRA ticket noting this branch was missed intentionally and make it blocked by the API ticket